src/Tests/dotnet.Tests/ParserTests/MSBuildArgumentCommandLineParserTests.cs

```diff
@@ -49,6 +49,9 @@ public void MSBuildArgumentsAreForwardedCorrectly(string[] arguments, bool build
         [InlineData(new string[] { "-p:teamcity_buildConfName=\"Build, Test and Publish\"" }, new string[] { "--property:teamcity_buildConfName=\"Build, Test and Publish\"" })]
         [InlineData(new string[] { "-p:prop1=true", "-p:prop2=false" }, new string[] { "--property:prop1=true", "--property:prop2=false" })]
         [InlineData(new string[] { "-p:prop1=\".;/opt/usr\"" }, new string[] { "--property:prop1=\".%3B/opt/usr\"" })]
+        [InlineData(new string[] { "-p:prop1=true;prop2=false;prop3=\"wut\";prop4=\"1;2;3\"" }, new string[]{ "--property:prop1=true", "--property:prop2=false", "--property:prop3=\"wut\"", "--property:prop4=\"1%3B2%3B3\""})]
+        [InlineData(new string[] { "-p:prop4=\"1;2;3\"" }, new string[]{ "--property:prop4=\"1%3B2%3B3\""})]
+        [InlineData(new string[] { "-p:prop4=\"1 ;2 ;3 \"" }, new string[]{ "--property:prop4=\"1 %3B2 %3B3 \""})]
         public void Can_pass_msbuild_properties_safely(string[] tokens, string[] forwardedTokens) {
```

Comment:
Might be helpful to add these tests and show it works with different aliases and also that it works with values wraped in ' instead of just ". That could also be overkill. 


     [InlineData(new string[] { "/p:BuildDir=\"C:\\users\\foo\\folder;\""}, new string[] {"--property:BuildDir=\"C:\\users\\foo\\folder%3B\""})]
        [InlineData(new string[] { "/property:BuildDir=\"C:\\users\\foo\\folder;\"" }, new string[] { "--property:BuildDir=\"C:\\users\\foo\\folder%3B\"" })]
        [InlineData(new string[] { "/property:BuildDir=\'C:\\users\\foo\\folder'" }, new string[] { "--property:BuildDir=\'C:\\users\\foo\\folder'" })]
        [InlineData(new string[] { "/p:Configuration='Release';" }, new string[] { "--property:Configuration='Release'" })]